src/Psalm/Internal/Analyzer/Statements/Expression/ArrayAnalyzer.php

```diff
@@ -333,19 +333,32 @@ private static function analyzeArrayItem(
                 } elseif ($key_type->isSingleIntLiteral()) {
                     $item_key_value = $key_type->getSingleIntLiteral()->value;
 
-                    if ($item_key_value >= $array_creation_info->int_offset) {
-                        if ($item_key_value === $array_creation_info->int_offset) {
+                    if ($item_key_value <= PHP_INT_MAX
```

Comment:
Ugh, I hate to think this could ever be false. Technically, TLiteralInt::$value is public and not typed so it could be a float higher than Max Int

I feel we should do something about that so this case would be handled upstream instead...

Reply:
I guess we can make it typed (in a separate PR) and see what happens.